app/src/main/java/org/schabi/newpipe/player/playqueue/PlayQueue.java

```diff
@@ -528,7 +528,16 @@ public boolean equals(@Nullable final Object obj) {
             return false;
         }
         final PlayQueue other = (PlayQueue) obj;
-        return streams.equals(other.streams);
+        if (size() != other.size()) {
+            return false;
+        }
+        for (int i = 0; i < size(); i++) {
+            // Check is based on URL
+            if (!streams.get(i).getUrl().equals(other.streams.get(i).getUrl())) {
```

Comment:
I know that the url should technically be enough to uniquely identify a stream, but in other places when comparing streams the service id is also used, so I'd add it.

Reply:
Added this check